spec/factories/miq_group.rb

```diff
@@ -5,17 +5,23 @@
     transient do
       features nil
       role nil
+
+      # Temporary: Allows entitlements, a new table, to be invisible to current spec setup code
+      # Do NOT use these if possible.
+      miq_user_role_id nil
+      miq_user_role nil
     end
 
     sequence(:sequence)  # don't want to spend time looking these up
     description { |g| g.role ? "EvmGroup-#{g.role}" : generate(:miq_group_description) }
 
     after :build do |g, e|
-      if e.role
-        g.miq_user_role = MiqUserRole.find_by_name("EvmRole-#{e.role}") ||
-                          FactoryGirl.create(:miq_user_role, :features => e.features, :role => e.role)
-      elsif e.features.present?
-        g.miq_user_role = FactoryGirl.create(:miq_user_role, :features => e.features)
+      if e.role || e.features || e.miq_user_role_id || e.miq_user_role
+        g.entitlement = FactoryGirl.create(:entitlement,
+                                           :features => e.features,
+                                           :role => e.role,
+                                           :miq_user_role_id => e.miq_user_role_id,
+                                           :miq_user_role => e.miq_user_role)
```

Comment:
This is cleaner (if you ignore the `miq_user_role_id`)
surprised this works actually.

I had to put this in because sometimes the features were not created yet. So it used existing groups (or roles) if it already existed in the db.

I have been trying to delete code that passed these around, so that is sweet that this simplified logic works.
very cool